pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ConsistentHashingStickyKeyConsumerSelector.java

```diff
@@ -89,18 +112,21 @@ public Consumer select(byte[] stickyKey) {
                 return null;
             }
 
-            Map.Entry<Integer, Consumer> ceilingEntry = hashRing.ceilingEntry(hash);
+            List<Consumer> consumerList;
+            Map.Entry<Integer, List<Consumer>> ceilingEntry = hashRing.ceilingEntry(hash);
             if (ceilingEntry != null) {
-                return ceilingEntry.getValue();
+                consumerList =  ceilingEntry.getValue();
             } else {
-                return hashRing.firstEntry().getValue();
+                consumerList = hashRing.firstEntry().getValue();
             }
+
+            return consumerList.get(hash % consumerList.size());
```

Comment:
@ltamber I have a question about this line of code in https://github.com/apache/pulsar/issues/23315#issuecomment-2360761086 . Are you available to answer that question? Thanks /cc @codelipenghui

Reply:
I made a change already, #23327